### PR TITLE
SISRP-7689, User::API controls nav tabs per delegate-view-as privileges

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.berkeley.edu/browse/SISRP-7689

Show/hide navigation according to delegate_permissions when user is in delegate-view-as mode. Tidying in `api.rb`, too.

### CODE REVIEW

File: spec/policies/authentication_state_spec.rb [hunk 1]
```diff
@@ -115,15 +115,14 @@
...
-      let(:campus_solutions_id) { '16777216' }
+      let(:campus_solutions_id) { '23009422' }
       let(:fake_session) {{
         'user_id' => user_id,
         'original_delegate_user_id' => random_id
       }}
       before do
-        crosswalk_proxy = CalnetCrosswalk::ByUid.new
+        allow(CalnetCrosswalk::ByUid).to receive(:new).with(user_id: user_id).and_return (crosswalk_proxy = double)
         allow(crosswalk_proxy).to receive(:lookup_campus_solutions_id).and_return campus_solutions_id
```
Comment (paulkerschen) [inline spec/policies/authentication_state_spec.rb]: @johncrossman FYI, did you know that `double` can take options for method stubbing? So these two lines could be expressed as

`allow(CalnetCrosswalk::ByUid).to receive(:new).with(user_id: user_id).and_return double(lookup_campus_solutions_id: campus_solutions_id)`

File: fixtures/xml/campus_solutions/delegate_access_students.xml [hunk 2]
```diff
@@ -3,20 +3,49 @@
...
+          <ROLENAME>UC DA Financial View</ROLENAME>
+          <ROLENAME>UC_CS_DA_ENROLLMENTS_GRADES_VW</ROLENAME>
+          <ROLENAME>UC DA InPerson Call Access</ROLENAME>
+        </ROLENAMES>
+      </STUDENT>
+      <STUDENT>
+        <EMPLID>24549574</EMPLID>
+        <NAME>Yap</NAME>
+        <ROLENAMES type="array">
+        </ROLENAMES>
```
Comment (paulkerschen) [inline fixtures/xml/campus_solutions/delegate_access_students.xml]: No roles, Yap!

File: app/models/user_specific_model.rb [hunk 3]
```diff
@@ -27,7 +27,7 @@ def directly_authenticated?
   end
 
   def delegate_permissions
-    @authentication_state.delegate_permissions || []
+    @authentication_state.delegate_permissions || nil
```
Comment (paulkerschen) [inline app/models/user_specific_model.rb]: @johncrossman `|| nil` here is probably superfluous?

**Comment by paulkerschen** [comment]
@johncrossman Things are looking well ordered for the Tulliver family. Thanks for the extra tidying in `api.rb`. I left a couple of minor style questions.

File: app/models/user/api.rb [hunk 4]
```diff
@@ -136,10 +138,26 @@ def is_delegate_user?
       @edo_attributes.present? && @edo_attributes[:delegate_user_id].present?
     end
 
+    def has_delegate_privilege(type)
+      return @delegate_privileges.present? && @delegate_privileges[type]
```
Comment (paulkerschen) [inline app/models/user/api.rb]: @johncrossman Unneeded `return`

**Comment by christianvuerings** [comment]
Old PR at #4808

File: spec/policies/authentication_state_spec.rb (hunk 1 above)
Comment (johncrossman) [inline spec/policies/authentication_state_spec.rb]: @paulkerschen nice. Amendment coming soon.

**Comment by johncrossman** [comment]
Amendments are in.

**Comment by paulkerschen** [comment]
Thanks @johncrossman.

**Pull request merged**